pkg/asset/agent/manifests/agentpullsecret.go

```diff
@@ -41,36 +43,53 @@ func (*AgentPullSecret) Name() string {
 // the asset.
 func (*AgentPullSecret) Dependencies() []asset.Asset {
 	return []asset.Asset{
+		&workflow.AgentWorkflow{},
+		&joiner.ClusterInfo{},
 		&agent.OptionalInstallConfig{},
 	}
 }
 
 // Generate generates the AgentPullSecret manifest.
 func (a *AgentPullSecret) Generate(dependencies asset.Parents) error {
+
+	agentWorkflow := &workflow.AgentWorkflow{}
 	installConfig := &agent.OptionalInstallConfig{}
-	dependencies.Get(installConfig)
-
-	if installConfig.Config != nil {
-		secret := &corev1.Secret{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: "v1",
-				Kind:       "Secret",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      getPullSecretName(installConfig),
-				Namespace: getObjectMetaNamespace(installConfig),
-			},
-			StringData: map[string]string{
-				pullSecretKey: installConfig.Config.PullSecret,
-			},
+	clusterInfo := &joiner.ClusterInfo{}
+	dependencies.Get(agentWorkflow, installConfig, clusterInfo)
+
+	switch agentWorkflow.Workflow {
+	case workflow.AgentWorkflowTypeInstall:
+		if installConfig.Config != nil {
+			a.generateSecret(installConfig.ClusterName(), installConfig.ClusterNamespace(), installConfig.Config.PullSecret)
 		}
-		a.Config = secret
 
+	case workflow.AgentWorkflowTypeAddNodes:
+		a.generateSecret(clusterInfo.ClusterID, clusterInfo.Namespace, clusterInfo.PullSecret)
```

Comment:
```suggestion
		a.generateSecret(clusterInfo.ClusterName, clusterInfo.Namespace, clusterInfo.PullSecret)
```
I think we can determine the ClusterName from APIDNSName and the installconfig's BaseDomain. It would be taking the APIDNSName and removing the BaseDomain and then removing the "api." prefix.

The ClusterName is a required parameter to import a cluster. Although the name can be anything random afaik, it would be nice to be consistent with how it is currently being set. Plus ClusterID is already a required parameter to import a cluster and having the name be the same as the ID could potentially be confusing.

Reply:
Thanks @rwsu , it looks like a better choice to use ClusterName. Done